### PR TITLE
fix(patch vue-router): url / to %2F encoding

### PR DESCRIPTION
**Pull request by fschade**
vue router v3* encodes slashes, pr fixes this temporary till we bump the router.

fixes: #5714

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsersSharingIndicator</strong> https://drone.owncloud.com/owncloud/web/18507/28/1
The following scenarios passed on retry:
- webUISharingInternalUsersSharingIndicator/shareWithUsers.feature:154

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationApp1</strong> https://drone.owncloud.com/owncloud/web/18507/65/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

File: packages/web-app-files/src/index.js [hunk 1]
```diff
@@ -322,6 +325,37 @@ const routes = [
 const translations = translationsJson
 const quickActions = quickActionsImport
 
+// type: patch
+// temporary patch till we have upgraded web to the latest vue router which make this obsolete
+// this takes care that routes like 'foo/bar/baz' which by default would be converted to 'foo%2Fbar%2Fbaz' stay as they are
+// should immediately go away and be removed after finalizing the update
+const patchRouter = router => {
+  // for now we only need the patch on files-personal route, if needed on more just add an d comment why
```
Comment (kulmann) [inline packages/web-app-files/src/index.js:333]: We also need it on the public files route :-)

File: packages/web-app-files/src/index.js [hunk 2]
```diff
@@ -194,8 +194,11 @@ const routes = [
     children: [
       {
         name: 'personal',
-        path: 'all/:item?/:page?',
```
Comment (kulmann) [inline packages/web-app-files/src/index.js:197]: Let's move the page to query params already. Will happen anyway.

File: packages/web-app-files/src/index.js [hunk 3]
```diff
@@ -194,8 +194,11 @@ const routes = [
...
+        path: 'all/:item*',
         component: Personal,
+        transform: () => {
+          console.log('111')
```
Comment (kulmann) [inline packages/web-app-files/src/index.js:200]: Trololo 👹

**Review by kulmann — CHANGES_REQUESTED** [review]

File: packages/web-app-files/src/index.js (hunk 3 above)
Comment (fschade) [reply packages/web-app-files/src/index.js:200]: 👾👾👾 its just there to check if the reviews are propper 🤣  i can remove it now

**Review by kulmann — COMMENTED** [review]
~~Hm.... could you also look into real blanks in URLs instead of `%20`? So that `New Folder` will be `New Folder` instead of `New%20Folder` in the URL?~~ 🤔

**Comment by fschade** [comment]
wait for https://github.com/owncloud/owncloud-design-system/pull/1626

**Comment by pascalwengerter** [comment]
@fschade please rebase on #5770 (motivation for the change of merge targets is outlined there). You can then use the ODS 10.0.0

File: changelog/unreleased/bugfix-url-encoding [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Bugfix: url encoding / decoding
+
+Fixes Vue router (prior to version 4) encoding from `files/all/folder` to `files%2Fall%2Ffolder`.
```
Comment (pascalwengerter) [inline changelog/unreleased/bugfix-url-encoding:3]: Didn't you fix it the other way around? 🤔

**Review by pascalwengerter — COMMENTED** [review]
We have an integration test for the pagination as well and I'm wondering why you didn't have to touch that...could you take a look?

**Comment by pascalwengerter** [comment]
Also, is this a bugfix or a breaking change for users (since, e.g., bookmarks don't work anymore as they should)?

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingInternalUsers2</strong> https://drone.owncloud.com/owncloud/web/18860/56/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingInternalUsers2</strong> https://drone.owncloud.com/owncloud/web/18879/56/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Review by kulmann — APPROVED** [review]
Awesome! 💪

**Pull request merged**